app/classes/message_expiration.rb

```diff
@@ -21,6 +21,12 @@ def initialize(limit, type)
     @type = type
   end
 
+  def wait_time
+    raise 'Only time based expirations have a wait time' unless time_based?
```

Comment:
define a custom error for this, wdyt?

```suggestion
    raise Messages::WaitTimeOnlyExpirationType unless time_based?
```

Reply:
Nice, Done!